### PR TITLE
[crypto] renaming NamespaceKeys -> namespace::Keys

### PR DESCRIPTION
**Pull request by mimoo**
This PR applies four changes all related to the same issue: naming in crypto.

1. Use namespaces to differentiate different algorithm keys (`ed25519::Ed25519PublicKey` becomes `ed25519::PublicKey` (see rational below).
2. Prefix specific crypto traits with `T` to avoid name collisions (because people like to import `traits::*` and we can't do `use traits::* as _` in rust yet.
3. Rename private/public key into signing/verifying key (this gives a good indication as to what type of keys we're handling).
4. Rename multi ed25519 signing/verifying "key" into signing/verifying "keys".

The biggest change is obviously the use of namespaces to indicate what types of keys we are using.
The rational is that:

* from the point of view of the ed25519 library, a key is just a key
* from the point of view of the caller, rust provides namespace just for this

### CODE REVIEW

**Comment by mimoo** [comment]
@matbd @thefallentree I don't think this is a breaking change, hence I didn't tag the PR with "breaking". Tell me if I'm wrong : o

**Comment by matbd** [comment]
I just want to say: please choose the best solution for the Rust codebase and don't worry too much about name collisions in serde-reflection / python (we can add support for namespaces very quickly if needed).

@davidiw @huitseeker @kchalkias

File: client/cli/src/client_proxy.rs [hunk 1]
```diff
@@ -748,8 +745,8 @@ impl ClientProxy {
     pub fn submit_signed_transaction(
         &mut self,
         raw_txn: RawTransaction,
-        public_key: Ed25519PublicKey,
-        signature: Ed25519Signature,
+        public_key: ed25519::VerifyingKey,
```
Comment (matbd) [inline client/cli/src/client_proxy.rs:748]: IMHO we should not break the consistency between variables and types. So, I would go for `public_key: XXXPublicKey` or `verifying_key: XXXVerifyingKey`

File: crypto/crypto/src/unit_tests/multi_ed25519_test.rs [hunk 2]
```diff
@@ -189,14 +192,14 @@ fn test_multi_ed25519_signature_serialization() {
     // Test 32 of 32
     test_successful_signature_serialization(&priv_keys_32, 32);
 
-    // Construct from single Ed25519Signature.
+    // Construct from single ed25519::Signature.
     let single_signature = priv_keys_3[0].sign_message(&MESSAGE_HASH);
-    let multi_signature = MultiEd25519Signature::from(single_signature.clone());
+    let multi_signature = multi_ed25519::VerifyingKeys::from(single_signature.clone());
```
Comment (huitseeker) [inline crypto/crypto/src/unit_tests/multi_ed25519_test.rs:197]:
```suggestion
    let multi_signature = multi_ed25519::MultiSignature::from(single_signature.clone());
```

Comment (mimoo) [reply crypto/crypto/src/unit_tests/multi_ed25519_test.rs:197]: good catch!

**Comment by mimoo** [comment]
Thanks for the fixing PR @huitseeker I'll do this.

I'll wait for your tool to catch up @matbd then : )

File: client/cli/src/client_proxy.rs (hunk 1 above)
Comment (mimoo) [reply client/cli/src/client_proxy.rs:748]: actually, in network we do call it a signing key:

```
    pub signing_keys: KeyPair<Ed25519PrivateKey>,
```

Comment (mimoo) [reply client/cli/src/client_proxy.rs:748]: (even signing "keys" :D)

**Comment by davidiw** [comment]
Can we continue the work here? Alternatively should we consider renaming x25519 back to X25519*?

**Comment by mimoo** [comment]
this is blocked by matthieu's tool, so probably not worth pursuing this rename at this point. We can rename `x25519::PublicKeys` to `x25519::X25519Publickeys` in https://github.com/libra/libra/pull/4229

**Pull request closed without merging**